src/core/io_queue.cc

```diff
@@ -348,14 +348,22 @@ class queued_io_request {
 
 class queued_read_write_request : public queued_io_request {
 private:
+    internal::io_sink& _sink;
     internal::io_request _io_request;
     std::unique_ptr<io_desc_read_write> _desc;
 
     bool is_cancelled() const noexcept { return !_desc; }
 
 public:
-    queued_read_write_request(internal::io_request req, io_queue& q, fair_queue_entry::capacity_t cap, io_queue::priority_class_data& pc, io_direction_and_length dnl, iovec_keeper iovs)
+    queued_read_write_request(internal::io_sink& sink,
+                              internal::io_request req,
+                              io_queue& q,
+                              fair_queue_entry::capacity_t cap,
+                              io_queue::priority_class_data& pc,
+                              io_direction_and_length dnl,
+                              iovec_keeper iovs)
```

Comment:
So now the read-write needs to reference
- io-queue
- io-sink
- proprity-class-data

:( Too messy already, but I don't suggest to do anything about it right now

Reply:
This PR introduced the dependency to `io_sink` -- now the request knows how it wants to be dispatched.

The remaining dependencies are preserved from the baseline.